sdk/eventhubs/azure_messaging_eventhubs/tests/force_errors.rs

```diff
@@ -0,0 +1,523 @@
+                            partition_id: Some(EVENTHUB_PARTITION.to_string()),
+                            partition_key: Some("My Partition Key.".to_string()),
+                            ..Default::default()
+                        }))
+                        .await
+                        .unwrap();
+
+                    for i in 1..200 {
+                        assert!(batch
+                            .try_add_event_data(
+                                EventData::builder()
+                                    .with_body(b"Hello, World!")
+                                    .add_property("Message#".to_string(), i)
+                                    .with_message_id(i)
+                                    .build(),
+                                None
+                            )
+                            .unwrap());
+                    }
+                    producer.send_batch(batch, None).await.unwrap()
+                }
+            }
+        },
+        |producer| {
+            producer
+                .force_error(azure_core::Error::new(
+                    azure_core::error::ErrorKind::Amqp,
+                    azure_core_amqp::AmqpError::from(AmqpErrorKind::SessionDetachedByRemote(
+                        Box::new(azure_core::error::Error::new(
+                            azure_core::error::ErrorKind::Other,
+                            "Forced error",
+                        )),
+                    )),
+                ))
+                .unwrap();
+        },
+        Duration::seconds(10), // Seconds until stable state.
+        Duration::seconds(30), // Seconds until test timeout.
+    )
+    .await?;
+
+    Ok(())
+}
+
+#[recorded::test(live)]
+async fn force_errors_send_batch_connection_error(ctx: TestContext) -> Result<()> {
+    const EVENTHUB_PARTITION: &str = "1";
+    const TEST_NAME: &str = "force_errors_send_batch_connection_error";
+    let recording = ctx.recording();
+    let host = recording.var("EVENTHUBS_HOST", None);
+    let eventhub = recording.var("EVENTHUB_NAME", None);
+    let credential = recording.credential();
+    let producer = Arc::new(
+        ProducerClient::builder()
+            .with_application_id(TEST_NAME.to_string())
+            .open(host.as_str(), eventhub.as_str(), credential.clone())
+            .await?,
+    );
+
+    force_errors(
+        producer.clone(),
+        |producer: Arc<ProducerClient>| {
+            let producer = producer.clone();
+            async move {
+                loop {
+                    let batch = producer
+                        .create_batch(Some(EventDataBatchOptions {
+                            partition_id: Some(EVENTHUB_PARTITION.to_string()),
+                            partition_key: Some("My Partition Key.".to_string()),
+                            ..Default::default()
+                        }))
+                        .await
+                        .unwrap();
+
+                    for i in 1..200 {
```

Comment:
Given the 10, 30, even 45 second time outs, is this sample *always* going to be enough? Should this async fn actually sleep itself to make sure the producer is forced into an error in time? If I'm following the logic correctly, this seems to assume a lot about modern hardware (which, of course, will change).

Reply:
The test puts 200 messages in a batch and then loops continuously until the future is dropped.

The 200 is just the size of the batch being sent (lines 79-90 add events to the batch, and 91 sends the batch.